wasm/execution/configuration.py

```diff
@@ -218,8 +214,11 @@ def __init__(self, store: Store) -> None:
     def execute(self) -> Tuple[TValue, ...]:
         from wasm.logic import OPCODE_TO_LOGIC_FN
 
-        while self.has_active_frame:
-            instruction = next(self.instructions)
+        while True:
+            try:
+                instruction = next(self.instructions)
+            except AttributeError:
+                break
 
             logic_fn = OPCODE_TO_LOGIC_FN[instruction.opcode]
```

Comment:
Looks like if there is an `AttributeError` right away, then `instruction` hasn't been assigned yet. Maybe that's fine if we have proven that impossible during a validation step? (although probably worth a comment, then)

Reply:
Maybe I'm mis-understanding here, but the `try/except` above breaks out of the loop on attribute error which should give us the strong guarantee that access to the local `instruction` variable can only happen in the case that the next instruction was pulled from the instruction sequence.

Reply:
Hmmm, must have been a brain-fart. I am guessing I thought it was on a different indentation. :+1: